### PR TITLE
[DRAFT] Mounting in C

### PR DESCRIPTION
**Pull request by flo91**
## Basics

early stage, work in progress, needs heavy refactoring, error handling not implemented yet,
**please do not review for now**

- [ ] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [x] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [ ] The PR is rebased with current master.

## Checklist

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation (see [Documentation Guidelines](https://www.libelektra.org/devgettingstarted/documentation))
- [ ] I fixed all affected decisions (see [Decision Process](https://www.libelektra.org/decisions/decision-process))
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://www.libelektra.org/devgettingstarted/coding))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned [every code](/.reuse/dep5) not directly written by me in [reuse syntax](https://reuse.software/)

## Review

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

- [x] Add the "work in progress" label if you do not want the PR to be reviewed yet.
- [ ] Add the "ready to merge" label **if the basics are fulfilled** and no further pushes are planned by you.

### CODE REVIEW

File: src/include/kdbmount.h [hunk 1]
```diff
@@ -0,0 +1,49 @@
+/* Mount API for newbackend, currently re-implementation of old mounting process in C instead of C++ */
+
+#ifndef KDB_MOUNT_H
+#define KDB_MOUNT_H
+
+#include <stdbool.h>
+#include <kdb.h>
+
+
+struct cBackendInfo {
```
Comment (markus2330) [inline src/include/kdbmount.h:10]: I don't think everything should start with c.

```suggestion
struct BackendInfo {
```

File: src/include/kdbmount.h [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+/* Linked List for BackendInfo structs */
+struct cListBackendInfo {
+	struct cBackendInfo backendInfo;
+	struct cListBackendInfo * next;
+};
+
+
+/* Actual mounting functionality */
+
+/* from mountbase.cpp */
```
Comment (markus2330) [inline src/include/kdbmount.h:26]: Mark this with TODO, as this should be removed later.

File: src/include/kdbmount.h [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+const KeySet * cReadMountConf (bool clNull, bool clFirst, bool clSecond, bool clThird, bool clVerbose, bool clDebug);
```
Comment (markus2330) [inline src/include/kdbmount.h:27]: Is this public API? Then it should start with a common prefix, e.g. elektraMount.

File: src/include/kdbmount.h [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+const char * cGetMountpoint (const KeySet * const mountconf, bool clInteractive);
+
+/* from mount.cpp */
+void cOutputMtab (KeySet * const mountConf, bool clFirst, bool clSecond, bool clNull);
```
Comment (markus2330) [inline src/include/kdbmount.h:31]: The API shouldn't output anything, please put this into the CLI.

File: src/libs/elektra/cmount.c [hunk 5]
```diff
@@ -0,0 +1,373 @@
...
+#include <kdbhelper.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
+
+
+
+
+/* Read mount configuration from the KDB
```
Comment (markus2330) [inline src/libs/elektra/cmount.c:11]: Please add full docu to public API.

**Review by markus2330 — CHANGES_REQUESTED** [review]
Good to see first code! I added a few comments.

**Review by flo91 — COMMENTED** [review]
Thanks for the useful notes!

**Comment by flo91** [comment]
The new PR #4851 contains is based on changes in the PR, so this PR represents a deprecated state and is now obsolete.
Additionally, instead of merging into master, the PR #4851 is for merging into `mount_cli_integration`.

**Pull request closed without merging**